### PR TITLE
[test] Adapt MCH regression tests to C2SM-RCM environments

### PR DESCRIPTION
**Pull request by ajocksch**
Addresses #499.

### CODE REVIEW

File: cscs-checks/mch/automatic_arrays.py [hunk 1]
```diff
@@ -8,11 +8,13 @@ def __init__(self):
...
         self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
         self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-pgi',
-                                    'PrgEnv-gnu']
+                                    'PrgEnv-gnu', 'PrgEnv-cray-c2sm-gpu',
+                                    'PrgEnv-pgi-c2sm-gpu']
         if self.current_system.name in ['daint', 'dom']:
             self.modules = ['craype-accel-nvidia60']
         elif self.current_system.name in ['kesch']:
             self.modules = ['craype-accel-nvidia35']
+            # workaround
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/automatic_arrays.py]: done

File: cscs-checks/mch/g2g_meteoswiss_check.py [hunk 2]
```diff
@@ -9,7 +9,10 @@ def __init__(self, g2g):
         self.descr = 'G2G Meteoswiss check with G2G=%s' % g2g
         self.strict_check = False
         self.valid_systems = ['kesch:cn']
-        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.valid_prog_environs = ['PrgEnv-gnu',
+# temporary workaround until the mvapich module is fixed
+#                                    'PrgEnv-gnu-c2sm-gpu',
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/g2g_meteoswiss_check.py]: done

File: cscs-checks/mch/gpu_direct_acc.py [hunk 3]
```diff
@@ -9,7 +9,11 @@ def __init__(self):
         super().__init__()
         self.descr = 'tests gpu-direct for Fortran OpenACC'
         self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
-        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-pgi']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-cray-c2sm-gpu',
+                                    'PrgEnv-pgi',
+# temporary workaround until the mvapich module is fixed
+#                                    'PrgEnv-pgi-c2sm-gpu',
+                                   ]
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/gpu_direct_acc.py]: done

File: cscs-checks/mch/gpu_direct_cuda.py [hunk 4]
```diff
@@ -9,7 +9,10 @@ def __init__(self):
         super().__init__()
         self.descr = 'tests gpu-direct for CUDA'
         self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
-        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.valid_prog_environs = ['PrgEnv-gnu',
+# temporary workaround until the mvapich module is fixed
+#                                    'PrgEnv-c2sm-gpu',
+                                   ]
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/gpu_direct_cuda.py]: done

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 5]
```diff
@@ -10,36 +10,48 @@ def __init__(self, withmpi):
...
         self.descr = 'test for OpenACC, CUDA, MPI, and C++'
         self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
-        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-pgi']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-pgi',
+                                    'PrgEnv-gnu',
+                                    'PrgEnv-cray-c2sm-gpu',
+# temporary workaround until the mvapich module is fixed
+#                                    'PrgEnv-pgi-c2sm-gpu',
+#                                    'PrgEnv-gnu-c2sm-gpu',
+                                   ]
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: done

File: cscs-checks/mch/src/openacc_cuda_mpi_cppstd.F90 [hunk 6]
```diff
@@ -70,9 +70,11 @@ program openacc_cuda_mpi_cppstd
     ! Call a CUDA kernel with host arrays 
     call call_cuda_kernel_with_copy(f1, f2, NSIZE)
 
+#ifdef _OPENACC
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/src/openacc_cuda_mpi_cppstd.F90]: for PrgEnv-gnu OpenACC needs to be deactivated; the CUDA kernel without copy expects arrays on the GPU provided by OpenACC

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 7]
```diff
@@ -65,5 +77,11 @@ def setup(self, partition, environ, **job_opts):
                     '-acc', '-ta:tesla:cc35,cuda8.0', '-lstdc++',
                     '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64',
                     '-lcublas', '-lcudart']
+        elif environ.name.startswith('PrgEnv-gnu'):
+            self.build_system.ldflags = ['-lstdc++']
+            if self.current_system.name == 'kesch':
+                self.build_system.ldflags += [
+                    '-L/global/opt/nvidia/cudatoolkit/8.0.61/lib64',
+                    '-lcublas', '-lcudart']
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: done

File: cscs-checks/mch/automatic_arrays.py [hunk 8]
```diff
@@ -43,16 +45,31 @@ def __init__(self):
...
+            },
+            'PrgEnv-gnu-c2sm-gpu': {
+                'daint:gpu': {'time': (7.0E-03, None, 0.15)},
+                'dom:gpu': {'time': (7.3E-03, None, 0.15)},
+                'kesch:cn': {'time': (6.5E-03, None, 0.15)},
+            },
+            'PrgEnv-pgi-c2sm-gpu': {
+                'daint:gpu': {'time': (6.4E-05, None, 0.15)},
+                'dom:gpu': {'time': (6.3E-05, None, 0.15)},
+                'kesch:cn': {'time': (1.4E-04, None, 0.15)},
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/automatic_arrays.py]: done

**Comment by vkarak** [comment]
@ajocksch I have updated the PR fine tuning the tests and adapted the configuration to use the installed C2SM-RCM environments. Can you have a look?

**Comment by ajocksch** [comment]
it looks good

**Pull request merged**